mnelab/model.py

```diff
@@ -131,22 +131,36 @@ def find_events(self, stim_channel, consecutive=True, initial_event=True,
     @data_changed
     def events_from_annotations(self):
         """Convert annotations to events."""
-        events, _ = mne.events_from_annotations(self.current["data"])
+        events, mappings = mne.events_from_annotations(self.current["data"])
+        # Mappings for annots are swapped {str: int} should be {int: str}
+        mappings = dict((v, k) for k, v in mappings.items())
         if events.shape[0] > 0:
             self.current["events"] = events
-            self.history.append("events, _ = "
-                                "mne.events_from_annotations(data)")
+            self.current["event_mappings"] = mappings
+            self.history.append("events, event_mappings = "
+                                "mne.events_from_annotations(data)"
+                                "mappings = dict((v, k) for k, v in mappings.items())")
 
     @data_changed
     def annotations_from_events(self):
         """Convert events to annotations."""
-        annots = mne.annotations_from_events(
-            self.current["events"],
-            self.current["data"].info["sfreq"]
-        )
+        if "event_mappings" in self.current:
+            mappings = self.current["event_mappings"]
+        else:
+            mappings = None
```

Comment:
One more optimization, this can be shortened to `mappings = self.current.get("event_mappings")`.

Reply:
I see. `self.current["variable"]` is only for assignment, right? I updated it.

Reply:
No, the `dict.get` method automatically returns `None` if the key is not present, so there's no need to check for `None`.